### PR TITLE
Remove finddomnode from hook

### LINKED ISSUE #73: Remove finddomnode from the useInViewport hook
You can easily get rid of the about to be deprecated finddomnode with react hooks:
[example](https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node)

### PR DESCRIPTION
**Pull request by Olierook**
Fixes #73, 

Also adds some explanation on the hook usage for #72  

The explanation is currently a comment in the hook file itself. It's up to you to add it in the README

### CODE REVIEW

**Comment by roderickhsiao** [comment]
Thanks for the PR. This will be a breaking changes for the below reasons
1. HOC will break
2. We will be requiring ref attached to component now
3. component using custom ref like
```js
const ref = useRef()
const {} = useInViewport();
const <div ref={ref} />
```
will have difficulty to adopt/migrate.

The other thoughts will be providing the fictionality but keep the existing findDOMNode as fallback for target

**Comment by Olierook** [comment]
So I've changed it to make 'getNode' optional. When provided with a target, the hook should work as normal. If a target is not set, the hook will try to set the node with the getNode function.

**Comment by roderickhsiao** [comment]
Thanks! looks good, I'll need sometime to think trough and test it out.

File: src/lib/useInViewport.js [hunk 1]
```diff
@@ -1,8 +1,52 @@
...
+
+*/
+import { useEffect, useRef, useState, useCallback, useMemo } from 'react';
 import { findDOMNode } from 'react-dom';
 
 const useInViewport = (target, options, config = { disconnectOnLeave: false }, props) => {
+  const [node, setNode] = useState(null);
+
+  // When target is provided, set node by finding DOM node.
+  useMemo(() => {
```
Comment (roderickhsiao) [inline src/lib/useInViewport.js:37]: guess this should be `useEffect` instead? `useMemo` store a function which returns value, and lazily executed when accessed.

**Comment by roderickhsiao** [comment]
fixed by #125

**Pull request closed without merging**